docs/DeveloperGuide.md

```diff
@@ -338,10 +338,47 @@ The following sequence diagram demonstrates how `StudyTrackerParser` parses the
 
 We felt that our choice would be the most intuitive behaviour of filter. 
 
-#### Future Extensions:
+#### Future Extensions
 
 A future extension would be for List to support filtering of amenities as well.
 
+### Enhanced Edit Command
+
+#### Overview
+
+The Edit Command is enhanced to support the removal of specific tags and amenities.
+
+#### Implementation
+
+The Edit Command is facilitated by two classes: `EditCommand.java` and `EditCommandParser.java`.
+
+Given below is an example usage scenario of removing a tag from a study spot and how the mechanism behaves.
+
+Step 1. The user executes `edit spot/com rt/noisy` to remove `noisy:Tag` from `com:StudySpot`.
+`StudyTrackerParser` class creates a `EditCommandParser` to parse the command. `EditCommandParser` then parses the 
+user input to find the `Name` object of the study spot using `ParserUtil#parseName(String name)`. An `EditStudySpotDescriptor`
+object is also created and since a tag is to be removed, `EditStudySpotDescriptor#setRemovedTags(Set<Tag> removedTags)`
+is called. Both the `Name` and `EditStudySpotDescriptor` objects are then passed as arguments to construct 
+an `EditCommand` object.
+
+Step 2. `LogicManager` executes the `EditCommand` object and calls 
+`EditCommand#createEditedStudySpot(StudySpot studySpotToEdit, EditStudySpotDescriptor editStudySpotDescriptor)` which
+creates an `StudySpot` object with the edited fields. `Model#setStudySpot(StudySpot target, StudySpot editedStudySpot)`
+and `Model#updateFilteredStudySpotList(Predicate<StudySpot> predicate)` is then called to update the model 
+in StudyTracker.
+
+The following sequence diagram demonstrates how `StudyTrackerParser` parses the command.
+
+![Edit RemoveTagSequenceDiagram](images/EditRemoveTagSequenceDiagram.png)
```

Comment:
Personally I think if you are going to include a diagram, you do not need to be so specific in your steps. In your steps, try and give a high level over view of what is going on instead

Reply:
Perhaps

Step 1. The user enters 'edit spot/com rt/noisy' to remove the tag noisy from the Com StudySPot
Step 2. Based on what is parsed, an EditCommand is created

or smth like that

Reply:
I agree with this! The diagram is really detailed, so the steps can afford to be simpler